samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerCount.java

```diff
@@ -94,51 +95,56 @@ public GroupByContainerCount(int containerCount) {
       return group(tasks);
     }
 
-    TaskAssignmentManager taskAssignmentManager = localityManager.getTaskAssignmentManager();
-    List<TaskGroup> containers = getPreviousContainers(taskAssignmentManager, tasks.size());
-    if (containers == null || containers.size() == 1 || containerCount == 1) {
-      log.info("Balancing does not apply. Invoking grouper.");
-      Set<ContainerModel> models = group(tasks);
-      saveTaskAssignments(models, taskAssignmentManager);
-      return models;
-    }
+    TaskAssignmentManager taskAssignmentManager =  new TaskAssignmentManager(localityManager.getConfig(), new MetricsRegistryMap());
```

Comment:
Why pass a no-op metrics registry map? Either it creates useful metrics and we need to emit them, or it doesn't and we should remove this param.

Reply:
In my next follow-up patch as a part of SEP-11, we'll be passing in the taskAssignments(static map) directly in the grouper API interface. So we will deprecate this API and this change will be no longer relevant and will be removed.